### ISSUE TITLE
[Request] Bot.logout

### ISSUE THREAD

**Issue by tcyrus** [TwitchIO/TwitchIO#97]
A safe logout function would be useful. ([example from discord.py](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.Bot.logout))

**Comment by IAmTomahawkx** [comment]
a logout function is not required, as there is no logging out from the twitch ws

**Comment by tcyrus** [comment]
Don't you still need to close the websocket connection?

**Comment by IAmTomahawkx** [comment]
that is handled when keyboardinterrupt is raised

**Comment by tcyrus** [comment]
One of the potential uses of a logout command is to safely shutdown the bot without the use of Errors.

**Comment by scragly** [comment]
If you use `commands.Bot.run` when starting your bot client, on exiting the script with almost any method will result in a safe websocket close. This is because in the `run` method, `self._ws.teardown` [is called within a `finally` block](https://github.com/TwitchIO/TwitchIO/blob/8a77d78049c726240f82385dda0930acafe4d9db/twitchio/ext/commands/bot.py#L309-L310). So you can safely call `sys.exit()` and the teardown will still be run, the websocket connection closed and everything cleanly exited without issue.

The exception is if you forcibly exit the process externally, or if you call explicitly `os._exit()` which is designed to stop the process in a way that bypasses any and all cleanup handling, including these `finally` blocks.

In Twitchio 2.0, we will consider if it's worth adding a `Bot.close` method. While there's no behaviour difference if you called it instead, we have a few people asking where a `Bot.close` method is, making it seem like a common expectation. The alternative is improving the documentation for the next version regarding closing the IRC connections to better explain the behaviour.

If any iteration comes to exist, it just won't likely be called `Bot.logout()`, as we don't have an API session to logout of, making the terminology inaccurate.

In summary, there's no need for a safe logout method as it's built to safely close the websocket when completing `Bot.run` or when exiting the script, so feel free to use `sys.exit()` for any of the ways you wish to exit without concern.

**Comment by Phil7789** [comment]
Maybe I am doing this completely wrong, but while trying to use sys.exit() to gracefully stop the script with a command I get a bunch of errors.

Using:
`@commands.command(name = 'shutdown')
    async def shutdown(self, ctx):
        await sys.exit(0)`

`**BASEPATH**/.local/lib/python3.6/site-packages/twitchio/websocket.py:618: RuntimeWarning: coroutine 'WebSocketCommonProtocol.close' was never awaited
  self._websocket.close()
Task exception was never retrieved
future: <Task finished coro=<Bot.event_message() done, defined at **BASEPATH**/.local/lib/python3.6/site-packages/twitchio/ext/commands/bot.py:767> exception=SystemExit(0,)>
Traceback (most recent call last):
  File "**BASEPATH**/.local/lib/python3.6/site-packages/twitchio/ext/commands/bot.py", line 306, in run
    loop.run_until_complete(self._ws._listen())
  File "/usr/lib/python3.6/asyncio/base_events.py", line 471, in run_until_complete
    self.run_forever()
  File "/usr/lib/python3.6/asyncio/base_events.py", line 438, in run_forever
    self._run_once()
  File "/usr/lib/python3.6/asyncio/base_events.py", line 1451, in _run_once
    handle._run()
  File "/usr/lib/python3.6/asyncio/events.py", line 145, in _run
    self._callback(*self._args)
  File "**BASEPATH**/.local/lib/python3.6/site-packages/twitchio/ext/commands/bot.py", line 777, in event_message
    await self.handle_commands(message)
  File "**BASEPATH**/.local/lib/python3.6/site-packages/twitchio/ext/commands/bot.py", line 524, in handle_commands
    await ctx.command._callback(instance, ctx, *ctx.args, **ctx.kwargs)
  File "**BASEPATH**/dev/twitch_bot/cogs/mods.py", line 8, in shutdown
    await sys.exit(0)
SystemExit: 0
`

**Comment by Gobot1234** [comment]
Thats cause you aren't meant to do that

**Comment by Phil7789** [comment]
Well what is an apropriate way of terminating the script during development using sys.exit() as stated above? I have to call it from somewhere (preferably from the chat if there are some odd behaviours)

**Comment by Gobot1234** [comment]
Try calling `Bot._ws.teardown()` first then exit

**Comment by jwgordon** [comment]
I am struggling with a similar issue.  How can I shut my bot down from inside the bot?  I have it running  remotely on a  server... it has a schedule that checks if it should be shut down, but I cant figure out how to  shut it down from inside.
